### PR TITLE
Add xschem and gaw

### PR DESCRIPTION
**Pull request by fedeinthemix**
###### Description of changes

Add schematic capture `xschem` and analog waveform viewer `gaw`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/xschem/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+stdenv.mkDerivation rec {
+
+  pname = "xschem";
+  version = "3.0.0";
+
+  src = fetchFromGitHub {
+      owner = "StefanSchippers";
+      repo = "xschem";
+      rev = "${version}";
+      sha256 = "sha256-C57jo8tAbiqQAgf4Xp2lpFGOr6F1knPpFcYxPiqSM4k=";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/xschem/default.nix:23]:
```suggestion
    owner = "StefanSchippers";
    repo = "xschem";
    rev = version;
    sha256 = "sha256-C57jo8tAbiqQAgf4Xp2lpFGOr6F1knPpFcYxPiqSM4k=";
```

File: pkgs/applications/science/electronics/xschem/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+, flex
+, libX11
+, libXpm
+, pkg-config
+, tcl
+, tk
+}:
+
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/xschem/default.nix:15]:
```suggestion
```

File: pkgs/applications/science/electronics/gaw/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "gaw";
+  version = "20200922";
+
+  src = fetchurl {
+    url = "http://download.tuxfamily.org/gaw/download/${pname}3-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/gaw/default.nix:14]:
```suggestion
    url = "https://download.tuxfamily.org/gaw/download/gaw3-${version}.tar.gz";
```

File: pkgs/applications/science/electronics/gaw/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ stdenv
+, fetchurl
+, lib
+, gtk3
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/gaw/default.nix:9]:
```suggestion
```

**Comment by fedeinthemix** [comment]
Thanks @SuperSandro2000 for the quick review and merging!

**Pull request merged**